RecoLocalTracker/SiPixelRecHits/plugins/PixelRecHits.cu

```diff
@@ -48,6 +48,7 @@ namespace pixelgpudetails {
     // Order such that the first ones are the ones transferred to CPU
     static_assert(sizeof(uint32_t) == sizeof(float)); // just stating the obvious
     cudaCheck(cudaMallocPitch(&gpu_.owner_32bit_, &gpu_.owner_32bit_pitch_, MAX_HITS*sizeof(uint32_t), 9));
+    cudaCheck(cudaMemsetAsync(gpu_.owner_32bit_, 0x0, gpu_.owner_32bit_pitch_*9, cudaStream.id()));
```

Comment:
why `pitch * 9` ?
shouldn't it be `pitch * 9 * MAX_HITS*sizeof(uint32_t)` ?

or am I just being confused ?

Reply:
ah, ok,  I was being confused: `pitch` is the rounded up value of `MAX_HITS*sizeof(uint32_t)`, so `pitch * 9` it is.

Reply:
`pitch` is the allocated width of a row in bytes
https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1g32bd7a39135594788a542ae72217775c